addon/components/g-map-polyline.js

```diff
@@ -0,0 +1,205 @@
+
+    mapContext.registerPolyline(this);
+  },
+
+  didInsertElement() {
+    this._super();
+    if (isEmpty(this.get('polyline'))) {
+      const options = compact(this.getProperties(allowedPolylineOptions));
+      const polyline = new google.maps.Polyline(options);
+      this.set('polyline', polyline);
+    }
+    this.setPath();
+    this.setMap();
+    this.setOnClick();
+  },
+
+  willDestroyElement() {
+    this.unsetPolylineFromMap();
+    this.get('mapContext').unregisterPolyline(this);
+  },
+
+  registerInfowindow(infowindow, openEvent, closeEvent) {
+    this.set('infowindow', infowindow);
+    this.attachOpenCloseEvents(infowindow, openEvent, closeEvent);
+  },
+
+  unregisterInfowindow() {
+    this.set('infowindow', null);
+  },
+
+  registerCoordinate(coordinate) {
+    this.get('coordinates').addObject(coordinate);
+  },
+
+  unregisterCoordinate(coordinate) {
+    this.get('coordinates').removeObject(coordinate);
+  },
+
+  attachOpenCloseEvents(infowindow, openEvent, closeEvent) {
+    const polyline = this.get('polyline');
+    if (openEvent === closeEvent) {
+      this.attachTogglingInfowindowEvent(polyline, infowindow, openEvent);
+    } else {
+      this.attachOpenInfowindowEvent(polyline, infowindow, openEvent);
+      this.attachCloseInfowindowEvent(polyline, infowindow, closeEvent);
+    }
+  },
+
+  attachOpenInfowindowEvent(polyline, infowindow, event) {
+    if (isPresent(event)) {
+      polyline.addListener(event, () => infowindow.open());
+    }
+  },
+
+  attachCloseInfowindowEvent(polyline, infowindow, event) {
+    if (isPresent(event)) {
+      polyline.addListener(event, () => infowindow.close());
+    }
+  },
+
+  attachTogglingInfowindowEvent(polyline, infowindow, event) {
+    if (isPresent(event)) {
+      polyline.addListener(event, () => {
+        if (infowindow.get('isOpen')) {
+          infowindow.close();
+        } else {
+          infowindow.open();
+        }
+      });
+    }
+  },
+
+  unsetPolylineFromMap() {
+    const polyline = this.get('polyline');
+    if (isPresent(polyline)) {
+      polyline.setMap(null);
+    }
+  },
+
+  mapWasSet: observer('map', function() {
+    run.once(this, 'setMap');
+  }),
+
+  setMap() {
+    const map = this.get('map');
+    const polyline = this.get('polyline');
+
+    if (isPresent(polyline) && isPresent(map)) {
+      polyline.setMap(map);
+    }
+  },
+
+  setPath() {
+    const polyline = this.get('polyline');
+    const coordinates = this.get('coordinates');
+    const options = compact(this.getProperties(allowedPolylineOptions));
+    var coordArray = [];
+    coordinates.forEach(function(coordinate){
+      const coord = coordinate.get('coordinate');
+      if (isPresent(coord)) {
+        coordArray.push(coord);
+      }
+    });
+    Ember.Logger.log(coordArray);
+    // assert('Must have at least two valid coordinates in {{#g-map-polyline}} component', coordArray.length > 1);
```

Comment:
I think we can remove that commented part, I agree that it's better to just do not set path, until we have 2 coords in array rather than spitting assertion error. Mostly, because coords may appear only after a few seconds.